### ISSUE TITLE
`test_pyrepl` may read its own output

### ISSUE THREAD

**Issue by colesbury** [python/cpython#121603]
First, a caveat: I don't understand psuedo-terminals well and I'm fairly confused by this.

When process start-up is slow (such as under heavy load), the pyrepl tests that use `test_pyrepl.run_repl` may read their own command as output. Data that is written to `master_fd` is immediately available to be read from `mater_fd`, so if the subprocess is not quick enough at reading the input.

https://github.com/python/cpython/blob/ef10110cd781afe8ddd3e65a54dc2b5ed2cb3187/Lib/test/test_pyrepl/test_pyrepl.py#L933

https://github.com/python/cpython/blob/ef10110cd781afe8ddd3e65a54dc2b5ed2cb3187/Lib/test/test_pyrepl/test_pyrepl.py#L948-L952

This leads to test errors that look like:

```
FAIL: test_python_basic_repl (test.test_pyrepl.test_pyrepl.TestMain.test_python_basic_repl)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/raid/sgross/cpython/Lib/test/support/__init__.py", line 2622, in wrapper
    return func(*args, **kwargs)
  File "/raid/sgross/cpython/Lib/test/test_pyrepl/test_pyrepl.py", line 892, in test_python_basic_repl
    self.assertIn("True", output)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
AssertionError: 'True' not found in 'from test.support import initialized_with_pyrepl\r\ninitialized_with_pyrepl()\r\nexit()\r\n'
```

In other words, the parent is seeing its own command as if its the output from the subprocess.

**Comment by colesbury** [comment]
I'm seeing this most frequently when I run lots of tests in parallel with TSan. TSan slows things down, which makes this more likely occur, but I don't think it's otherwise related to TSan.

**Comment by colesbury** [comment]
I think that I misidentified the problem. The reading it's own output seems to be expected (?). The actual problem is:

* the timeout is too short (0.5 seconds)
* increasing the timeout slows down the test (bad!)
